services/motion/builtin/builtin_test.go

```diff
@@ -1731,25 +1532,588 @@ func TestMoveOnGlobe(t *testing.T) {
+		ms.(*builtIn).fsService = fsSvc
+
+		req := motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   spatialmath.NewPoseFromPoint(r3.Vector{X: 1001, Y: 1001}),
+			SlamName:      slam.Named("test_slam"),
+			Extra:         map[string]interface{}{"timeout": 0.01},
+		}
+
+		executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+	})
+
+	t.Run("Changes to executions show up in PlanHistory", func(t *testing.T) {
+		kb, ms := createMoveOnMapEnvironment(ctx, t, "pointcloud/octagonspace.pcd", 40, nil)
+		defer ms.Close(ctx)
+		easyGoalInBaseFrame := spatialmath.NewPoseFromPoint(r3.Vector{X: 0.277 * 1000, Y: 0.593 * 1000})
+		easyGoalInSLAMFrame := spatialmath.PoseBetweenInverse(motion.SLAMOrientationAdjustment, easyGoalInBaseFrame)
+
+		req := motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   easyGoalInSLAMFrame,
+			SlamName:      slam.Named("test_slam"),
+			Extra:         map[string]interface{}{"smooth_iter": 0},
+		}
+
+		executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotBeEmpty)
+
+		// returns the execution just created in the history
+		ph, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(ph), test.ShouldEqual, 1)
+		test.That(t, ph[0].Plan.ExecutionID, test.ShouldResemble, executionID)
+		test.That(t, len(ph[0].StatusHistory), test.ShouldEqual, 1)
+		test.That(t, ph[0].StatusHistory[0].State, test.ShouldEqual, motion.PlanStateInProgress)
+		test.That(t, len(ph[0].Plan.Steps), test.ShouldNotEqual, 0)
+
+		err = ms.StopPlan(ctx, motion.StopPlanReq{ComponentName: kb.Name()})
+		test.That(t, err, test.ShouldBeNil)
+
+		ph2, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(ph2), test.ShouldEqual, 1)
+		test.That(t, ph2[0].Plan.ExecutionID, test.ShouldResemble, executionID)
+		test.That(t, len(ph2[0].StatusHistory), test.ShouldEqual, 2)
+		test.That(t, ph2[0].StatusHistory[0].State, test.ShouldEqual, motion.PlanStateStopped)
+		test.That(t, ph2[0].StatusHistory[1].State, test.ShouldEqual, motion.PlanStateInProgress)
+		test.That(t, len(ph2[0].Plan.Steps), test.ShouldNotEqual, 0)
+
+		// Proves that calling StopPlan after the plan has reached a terminal state is idempotent
+		err = ms.StopPlan(ctx, motion.StopPlanReq{ComponentName: kb.Name()})
+		test.That(t, err, test.ShouldBeNil)
+		ph3, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, ph3, test.ShouldResemble, ph2)
+	})
+
+	t.Run("Returns error when called with an unknown component", func(t *testing.T) {
```

Comment:
the remaining tests below copy what is already tested in `TestMoveOnGlobe`

Reply:
All these tests that start with "Returns error" are kind of just input validation tests, and IMO don't belong in the same `TestThing()` function as the integration tests above that actually call a motion planner.

Could we move the "Returns error" tests for both MoveOnMap, and MoveOnGlobe, into a new `TestMoveCallInputs` function?

They should be able to run in parallel that way as well.

Reply:
great idea!